File: legacy_python/app/models/basic.py
import mongoengine as db
from datetime import datetime

class DailyTask(db.Document):
    name = db.StringField(required=True, max_length=100)
    default_stars = db.IntField(required=True)
    description = db.StringField(max_length=200)

class Student(db.Document):
    name = db.StringField(required=True, max_length=100)
    total_stars = db.IntField(default=0)
    # Rewards are referenced back or embedded? 
    # For now, stick to Referenced to maintain structure similar to SQL
    
    meta = {'strict': False}

class Reward(db.Document):
    stars = db.IntField(required=True)
    reason = db.StringField(required=True, max_length=200)
    date = db.DateTimeField(default=datetime.utcnow)
    student = db.ReferenceField(Student, required=True)
    is_penalty = db.BooleanField(default=False)
    task = db.ReferenceField(DailyTask)

class Task(db.Document):
    student = db.ReferenceField(Student, required=True)
    title = db.StringField(required=True, max_length=100)
    description = db.StringField(max_length=200)
    due_date = db.DateTimeField(required=True)
    is_completed = db.BooleanField(default=False)

File: src/models/Basic.js
const mongoose = require('mongoose');

const dailyTaskSchema = new mongoose.Schema({
    name: { type: String, required: true, maxlength: 100 },
    default_stars: { type: Number, required: true },
    description: { type: String, maxlength: 200 }
});

const studentSchema = new mongoose.Schema({
    name: { type: String, required: true, maxlength: 100 },
    total_stars: { type: Number, default: 0 }
}, { strict: false });

const rewardSchema = new mongoose.Schema({
    stars: { type: Number, required: true },
    reason: { type: String, required: true, maxlength: 200 },
    date: { type: Date, default: Date.now },
    student: { type: mongoose.Schema.Types.ObjectId, ref: 'Student', required: true },
    is_penalty: { type: Boolean, default: false },
    task: { type: mongoose.Schema.Types.ObjectId, ref: 'DailyTask' }
});

const taskSchema = new mongoose.Schema({
    student: { type: mongoose.Schema.Types.ObjectId, ref: 'Student', required: true },
    title: { type: String, required: true, maxlength: 100 },
    description: { type: String, maxlength: 200 },
    due_date: { type: Date, required: true },
    is_completed: { type: Boolean, default: false }
});

module.exports = {
    DailyTask: mongoose.model('DailyTask', dailyTaskSchema),
    Student: mongoose.model('Student', studentSchema),
    Reward: mongoose.model('Reward', rewardSchema),
    Task: mongoose.model('Task', taskSchema)
};
